passes/source2il.nim

```diff
@@ -28,6 +28,11 @@ type
 
   ModuleCtx* = object
     ## The translation/analysis context for a single module.
+    reporter: ref ReportContext[string]
+    # XXX: strings being used for messages is a temporary measure. Eventually,
+    #      a dedicated message type that offloads message formating to message
+    #      rendering will be used
```

Comment:
Instead of 'messages', I'd prefer the term 'diagnostics' and its derivatives. 'Messages' is very general, think 'help message', and better reserved for something else

In my head at least:
- **diagnostics** are something emit by the compiler/app core (there are the errors, warnings, hints, etc)
- more framework/infrastructure code, and the above compiler/app core diagnostics, are **logged** (loggers)
- both the compiler/app + infra can produce:
   - **telemetry** (metrics, for tracing, etc), or
   - plain **messages** to various **output channels** (typically stdout vs stderr, but could be files)

It's not a super clean hierarchy, but I hope it offers some insight. This has been informed pretty heavily by working on NimSkull and with the VS Code Extension API. The latter I've found maps fairly well to compiler use cases. The compiler core for example is embedded in the compiler app/infra, LSP, REPL, etc, this "core in a box" model kinda how VS Code approaches things.

N.B.: plain messages do not need to be, nor should they be, structured like other messages, the over unification/abstraction of NimSkull reports was crippling. Sometimes you just want to output a string. 😅

Reply:
> Instead of 'messages', I'd prefer the term 'diagnostics' and its derivatives. 'Messages' is very general, think 'help message', and better reserved for something else

I agree. 'message' is a very broad term, and the terms meaning in object-oriented programming makes its usage here even more confusing. I'll change it to 'diagnostic'.